source/val/validate_decorations.cpp

```diff
@@ -795,6 +795,15 @@ spv_result_t CheckDecorationsOfBuffers(ValidationState_t& vstate) {
         MemberConstraints constraints;
         ComputeMemberConstraintsForStruct(&constraints, id, LayoutConstraints(),
                                           vstate);
+
+        if (push_constant && !hasDecoration(id, SpvDecorationBlock, vstate)) {
```

Comment:
This check should also be gated by `spvIsVulkanEnv(vstate.context()->target_env)`. It does not apply generally to SPIR-V.